modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/ReindexNodeShutdownIT.java

```diff
@@ -0,0 +1,145 @@
+        assertHitCount(prepareSearch(INDEX).setSize(0).setTrackTotalHits(true), numDocs);
+    }
+
+    private void createReindexTaskAndShutdown(final String coordNodeName) throws Exception {
+        AbstractBulkByScrollRequestBuilder<?, ?> builder = reindex(coordNodeName).source(INDEX).destination(DEST_INDEX);
+        AbstractBulkByScrollRequest<?> reindexRequest = builder.request();
+        ShutdownPrepareService shutdownPrepareService = internalCluster().getInstance(ShutdownPrepareService.class, coordNodeName);
+
+        TaskManager taskManager = internalCluster().getInstance(TransportService.class, coordNodeName).getTaskManager();
+
+        // Now execute the reindex action...
+        ActionListener<BulkByScrollResponse> reindexListener = new ActionListener<BulkByScrollResponse>() {
+            @Override
+            public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
+                assertNull(bulkByScrollResponse.getReasonCancelled());
+                logger.debug(bulkByScrollResponse.toString());
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.debug("Encounterd " + e.toString());
+                fail(e, "Encounterd " + e.toString());
+            }
+        };
+        internalCluster().client(coordNodeName).execute(ReindexAction.INSTANCE, reindexRequest, reindexListener);
+
+        // Check for reindex task to appear in the tasks list and Immediately stop coordinating node
+        findTask(ReindexAction.INSTANCE.name(), coordNodeName);
+        shutdownPrepareService.prepareForShutdown(taskManager);
+        internalCluster().stopNode(coordNodeName);
+    }
+
+    // Make sure all documents from the source index have been reindexed into the destination index
+    private void checkDestinationIndex(String dataNodeName, int numDocs) {
+        assertTrue(indexExists(DEST_INDEX));
+        flushAndRefresh(DEST_INDEX);
+        assertTrue("Number of documents in source and dest indexes does not match", waitUntil(() -> {
+            final TotalHits totalHits = SearchResponseUtils.getTotalHits(
+                client(dataNodeName).prepareSearch(DEST_INDEX).setSize(0).setTrackTotalHits(true)
+            );
+            return totalHits.relation() == TotalHits.Relation.EQUAL_TO && totalHits.value() == numDocs;
+        }, 10, TimeUnit.SECONDS));
+    }
+
+    private static void findTask(String actionName, String nodeName) throws Exception {
+        assertBusy(() -> {
+            ListTasksResponse tasks = clusterAdmin().prepareListTasks(nodeName).setActions(actionName).setDetailed(true).get();
+            tasks.rethrowFailures("Find my task");
+            for (TaskInfo taskInfo : tasks.getTasks()) {
+                // Skip tasks with a parent because those are children of the task we want
+                assertFalse(taskInfo.parentTaskId().isSet());
+            }
```

Comment:
I think this does not really wait. As a sidenote I am unsure if we need to wait and also whether skipping child tasks is important. But I think this should be:

```suggestion
            for (TaskInfo taskInfo : tasks.getTasks()) {
                // Skip tasks with a parent because those are children of the task we want
                if(taskInfo.parentTaskId().isSet()) == false) {
                    return;
                }
                fail("not found - yet");
            }
```

Reply:
You are right. Thanks for catching this. But I actually don't like using assertBusy() here because it waits for an increasing amount of time and even the first wait is 1 ms. So I do in fact want to change this back to what I had originally where we wait for 10 nanoseconds at a time.

Reply:
You are right. Thanks for catching this, I fixed it and tested it out as well (by holding off submission of the reindexing task).